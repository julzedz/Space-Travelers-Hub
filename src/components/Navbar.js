import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/navbar.css';
import logo from './images/planet.png';

function Navbar() {
  return (
    <header className="nav-div">
      <section className="logo-div">
        <img className="space-hub-logo" src={logo} alt="Space Hub Logo" />
        <h1 className="logo">Space Travelers&apos; Hub</h1>
      </section>
      <nav className="nav-links">
        <Link className="links" to="/">Rockets</Link>
        <Link className="links" to="/missions">Missions</Link>
        <Link className="links" to="/profile">My Profile</Link>
      </nav>
    </header>
  );
}

export default Navbar;
