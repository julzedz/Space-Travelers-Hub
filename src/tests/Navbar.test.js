import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  test('correct render of Navbar', () => {
    const navbar = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(navbar).toMatchSnapshot();
  });
  test("Check for 'My Profile' in the rendered Navbar", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(screen.getByText('My Profile')).toBeInTheDocument();
  });
});
