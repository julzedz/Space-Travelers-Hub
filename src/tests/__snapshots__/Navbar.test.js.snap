// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar correct render of Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="nav-div"
      >
        <section
          class="logo-div"
        >
          <img
            alt="Space Hub Logo"
            class="space-hub-logo"
            src="planet.png"
          />
          <h1
            class="logo"
          >
            Space Travelers' Hub
          </h1>
        </section>
        <nav
          class="nav-links"
        >
          <a
            activeclassname="active-link"
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
          <a
            activeclassname="active-link"
            class=""
            href="/missions"
          >
            Missions
          </a>
          <a
            activeclassname="active-link"
            class=""
            href="/profile"
          >
            My Profile
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="nav-div"
    >
      <section
        class="logo-div"
      >
        <img
          alt="Space Hub Logo"
          class="space-hub-logo"
          src="planet.png"
        />
        <h1
          class="logo"
        >
          Space Travelers' Hub
        </h1>
      </section>
      <nav
        class="nav-links"
      >
        <a
          activeclassname="active-link"
          aria-current="page"
          class="active"
          href="/"
        >
          Rockets
        </a>
        <a
          activeclassname="active-link"
          class=""
          href="/missions"
        >
          Missions
        </a>
        <a
          activeclassname="active-link"
          class=""
          href="/profile"
        >
          My Profile
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
